tiledb/sm/cpp_api/dimension.h

```diff
@@ -247,6 +247,7 @@ class Dimension {
         // Strings have null domains so let's return an empty string
         // representation
         return "";
+      case TILEDB_BYTE:
```

Comment:
Arbitrary byte-sequences could be supported as a domain. Mathematically, it would have a "discrete topology", meaning every point was "at the same remove from" every other point. If we have no application in mind for them, however, that's a different question. Regardless, the documentation should reflect this choice.